### DISCUSSION TITLE
PocketBase won't create new user.

### DISCUSSION THREAD

**Discussion by kenyonbowers** [pocketbase/pocketbase#1315] [Q&A]
So I have some code,
`await client.collection('users').create({
            username: "Random Name",
            password: "MyPassword",
            passwordConfirm: "MyPassword",
            email: "test@example.com"
        });`
And when I try and run it, I get,
`ClientResponseError 400: Failed to create record.`
And I tried editing the Api Rules so anyone can create, update, and delete records with no limitations yet that still didn't allow me to create a new user(auth) record.

**Comment by ganigeorgiev** [comment]
The username is not in the correct format.
Usually the `ClientResponseError` should have `err.data` prop with details about the invalid fields.

You can also check the logs in the Admin UI > Logs and/or start PocketBase with `--debug`.

**Reply by kenyonbowers** [reply]
this is the full error
`ClientResponseError 400: Failed to create record.
    at new i2 (ClientResponseError.ts:13:9)
    at Client.ts:270:23
    at async Proxy.signup (UserService.ts:22:9)`

**Reply by ganigeorgiev** [reply]
@kenyonbwrs This is just the error object but it has some additional fields, including `err.data` (for more info please check https://github.com/pocketbase/js-sdk#error-handling).

**Comment by kenyonbowers** [comment]
also, according to the docs, the only required values are `password` and `confirmPassword`. And I also don't have any fields so that cant be the problem. I also removed `username` and `email` from the code because they are not required yet it still doesn't work.

**Reply by ganigeorgiev** [reply]
What version of PocketBase you are using?

Please check the Admin UI > Logs and/or start the executable with `--debug`.

**Reply by kenyonbowers** [reply]
Whatever the newest version is. I just set it up last night

**Reply by kenyonbowers** [reply]
PocketBase v0.10.1

**Reply by ganigeorgiev** [reply]
I've just tested it locally and I'm not able to reproduce the error. Please check the detailed error response (if the request is executed in a browser, you can also find the response in the browser's "Inspect tool > Network > find the 400 request > Response").

**Reply by kenyonbowers** [reply]
It all came down to a misspelling in the password field. I had retyped it here so the problem isn't present with the code above

**Comment by Doomd** [comment]
Speaking of `username` format, what IS the exact requirement for the username, and where can I find this documentation? It took me a while to figure out that a "Failed to create record" error I was having was due to a username having a "-" character for example. I also discovered that "+" signs are no good either. I couldn't find documentation anywhere on the required formats for system fields...

**Reply by ganigeorgiev** [reply]
I'll consider adding in the documentation sometime later.
The `username` string must satisfy the `^[\w][\w\.]*$` regex (it could change in the future).

_Just for info, you can access the individual field validation errors by inspecting the API error response available under the `err.data` object._

**Reply by Doomd** [reply]
thanks, that regex is just what I needed.
